src/lib.rs

```diff
@@ -240,15 +314,13 @@ pub fn parse_datetime_at_date<S: AsRef<str> + Clone>(
     // offsets, so instead we replicate parse_date behaviour by getting
     // the current date with local, and create a date time string at midnight,
     // before trying offset suffixes
-    let ts = format!("{}0000{}", date.format("%Y%m%d"), s);
-    for fmt in [
-        format::UTC_OFFSET,
-        format::ZULU_OFFSET,
-        format::NAKED_OFFSET,
-    ] {
-        let f = format::YYYYMMDDHHMM.to_owned() + fmt;
-        if let Ok(parsed) = DateTime::parse_from_str(&ts, &f) {
-            return Ok(parsed);
+    let ts = format!("{}0000{}", date.format("%Y%m%d"), tmp_s.as_ref());
+    for (fmt, n) in format::PATTERNS_OFFSET {
+        if ts.len() == n + 12 {
+            let f = format::YYYYMMDDHHMM.to_owned() + fmt;
+            if let Ok(parsed) = DateTime::parse_from_str(&ts[0..n + 12], &f) {
```

Comment:
Because of the condition on line 319 you can simplify it to:
```suggestion
            if let Ok(parsed) = DateTime::parse_from_str(&ts, &f) {
```